modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authorization/authorizer/rdbms/RdbmsAuthorizer.java

```diff
@@ -0,0 +1,175 @@
+        userCache = CacheBuilder.newBuilder()
+                                .maximumSize(brokerAuthConfiguration.getAuthorization().getCache()
+                                                                    .getSize())
+                                .expireAfterWrite(brokerAuthConfiguration.getAuthorization()
+                                                                         .getCache()
+                                                                         .getTimeout(),
+                                                  TimeUnit.MINUTES)
+                                .build(new UserCacheLoader());
+    }
+
+    @Override
+    public boolean authorize(String scopeName, String userId)
+            throws BrokerAuthException, BrokerAuthServerException, BrokerAuthNotFoundException {
+        try {
+            if (userId != null) {
+                UserCacheEntry userCacheEntry = userCache.get(userId);
+                boolean anyMatch = userCacheEntry.getAuthorizedScopes()
+                                                 .stream()
+                                                 .anyMatch(s -> s.equals(scopeName));
+                if (anyMatch) {
+                    LOGGER.debug("Scopes are loaded from cache for auth scope key : {} ", scopeName);
+                    return true;
+                } else {
+                    if (authScopeStore.authorize(scopeName, userCacheEntry.getUserGroups())) {
+                        userCacheEntry.getAuthorizedScopes().add(scopeName);
+                        return true;
+                    } else {
+                        return false;
+                    }
+                }
+            } else {
+                throw new BrokerAuthException("user id cannot be null.");
+            }
+        } catch (ExecutionException e) {
+            throw new BrokerAuthServerException("Error occurred while retrieving authorizations from cache for"
+                                                        + " scope name : " + scopeName, e);
+        }
+    }
+
+    @Override
+    public boolean authorize(String resourceType, String resourceName, String action, String userId)
+            throws BrokerAuthException, BrokerAuthServerException, BrokerAuthNotFoundException {
+        ResourceCacheKey resourceCacheKey = new ResourceCacheKey(resourceType, resourceName);
+        try {
+            if (userId != null) {
+                UserCacheEntry userCacheEntry = userCache.get(userId);
+                Set<String> authorizedActions =
+                        userCacheEntry.getAuthorizedResourceActions().get(resourceCacheKey);
+                boolean anyMatch = Objects.nonNull(authorizedActions) &&
+                        authorizedActions
+                                .stream()
+                                .anyMatch(s -> s.equals(action));
+                if (anyMatch) {
+                    LOGGER.debug("resourceName authorizations are loaded from cache for resourceType : " +
```

Comment:
It is better if we use an isDebugEnabled check or use string formatting method with curly braces in log4j.

Reply:
Fixed in 885868b